registry/registry.go

```diff
@@ -29,6 +30,9 @@ import (
 // used in a Deployment Manager configuration. There can be multiple
 // registry implementations.
 type Registry interface {
+	// Also handles http.Client.Do method for authenticated File accesses
+	util.HTTPDoer
```

Comment:
Well, that does feel a bit safer/better. Would prefer to see the chart/template format abstraction exposed, so that instead of doing raw HTTP against the URLs, the client would ask for things in the chart/template, make using relative names, and the registry would go get them.